### PR TITLE
feat: dependency set refactor

### PR DESCRIPTION
**Pull request by agusduha**

### CODE REVIEW

File: specs/interop/dependency-set.md [hunk 1]
```diff
@@ -46,20 +44,6 @@ changes, all possible OP Stack chains in the superchain will be able to exist co
 
 It is a known issue that not all software in the Ethereum can handle 32 byte chain ids.
 
-## Updating the Dependency Set
-
-The `SystemConfig` is updated to have the `SuperchainConfig` address and to manage the dependency set.
```
Comment (gotzenx) [inline specs/interop/dependency-set.md:51]: maybe answered later on the PR: why are we not adding that the SuperchainConfig is not in charge of this?

File: specs/protocol/superchain-configuration.md [hunk 2]
```diff
@@ -97,62 +96,51 @@ The contract will add the following storage layout and function:
...
-It can only be called by the `UPGRADER` role in the `SuperchainConfig` and ensures that the chain ID
+It can only be called by the `DEPENDENCY_MANAGER` role in the `SuperchainConfig` and ensures that the chain ID
 is not already included in the dependency set.
 
-Before proceeding, it verifies that the new chain's dependency set size is zero.
-The function updates all chain dependencies by executing deposit transactions to form a complete mesh graph,
-stores the provided `SystemConfig` address in the `systemConfigs` mapping, and allowlists the
-new chain's `OptimismPortal` in the `SharedLockbox`.
+Before proceeding, it verifies that the dependency set is not full (`type(uint8).max`).
+The function allowlists the new chain's `OptimismPortal` in the `SharedLockbox`.
```
Comment (gotzenx) [inline specs/protocol/superchain-configuration.md:117]:
```suggestion
This function allowlists the new chain's `OptimismPortal` in the `SharedLockbox`.
```

File: specs/protocol/superchain-configuration.md [hunk 3]
```diff
@@ -97,62 +96,51 @@ The contract will add the following storage layout and function:
...
 
-- Only the `UPGRADER` role MUST be able to add a new chain to the dependency set
-
-- The chain being added MUST NOT have any other dependencies before joining a cluster
+- Only the `DEPENDENCY_MANAGER` role MUST be able to add a new dependency to the dependency set
 
-- The same chain MUST NOT be added more than once
-
-- It MUST add the new chain as a dependency on each chain of the cluster
+- The dependency set MUST NOT be at it's max size before adding a new dependency
```
Comment (gotzenx) [inline specs/protocol/superchain-configuration.md:139]:
```suggestion
- A new chain cannot be added to the dependency set once the maximum size is reached.
```

File: specs/protocol/superchain-configuration.md [hunk 4]
```diff
@@ -97,62 +96,51 @@ The contract will add the following storage layout and function:
...
 
-- It MUST add the whole cluster as dependency set for the new chain
+- The chain ID MUST be different to the current `block.chainid`
```
Comment (gotzenx) [inline specs/protocol/superchain-configuration.md:141]:
```suggestion
- The chain ID MUST be different from the current `block.chainid`
```

File: specs/interop/dependency-set.md (hunk 1 above)
Comment (agusduha) [reply specs/interop/dependency-set.md:51]: We can add a line saying that and linking to the SuperchainConfig section

**Review by gotzenx — APPROVED** [review]

**Pull request merged**